File: stripe/src/test/java/com/stripe/android/model/ModelUtilsTest.java
package com.stripe.android.model;

import org.junit.Test;

import java.util.Calendar;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertNotEquals;
import static org.junit.Assert.assertTrue;

/**
 * Test class for {@link ModelUtils}.
 */
public class ModelUtilsTest {

    @Test
    public void wholePositiveNumberShouldFailNull() {
        assertFalse(ModelUtils.isWholePositiveNumber(null));
    }

    @Test
    public void wholePositiveNumberShouldPassIfEmpty() {
        assertTrue(ModelUtils.isWholePositiveNumber(""));
    }

    @Test
    public void wholePositiveNumberShouldPass() {
        assertTrue(ModelUtils.isWholePositiveNumber("123"));
    }

    @Test
    public void wholePositiveNumberShouldPassWithLeadingZero() {
        assertTrue(ModelUtils.isWholePositiveNumber("000"));
    }

    @Test
    public void wholePositiveNumberShouldFailIfNegative() {
        assertFalse(ModelUtils.isWholePositiveNumber("-1"));
    }

    @Test
    public void wholePositiveNumberShouldFailIfLetters() {
        assertFalse(ModelUtils.isWholePositiveNumber("1a"));
    }

    @Test
    public void normalizeSameCenturyShouldPass() {
        Calendar now = Calendar.getInstance();
        int year = 1997;
        now.set(Calendar.YEAR, year);
        assertEquals(ModelUtils.normalizeYear(97, now), year);
    }

    @Test
    public void normalizeDifferentCenturyShouldFail() {
        Calendar now = Calendar.getInstance();
        int year = 1997;
        now.set(Calendar.YEAR, year);
        assertNotEquals(ModelUtils.normalizeYear(97, now), 2097);
    }
}

File: payments-core/src/test/java/com/stripe/android/model/ModelUtilsTest.kt
package com.stripe.android.model

import java.util.Calendar
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertFalse
import kotlin.test.assertNotEquals
import kotlin.test.assertTrue

/**
 * Test class for [ModelUtils].
 */
class ModelUtilsTest {

    @Test
    fun wholePositiveNumberShouldFailNull() {
        assertFalse(ModelUtils.isWholePositiveNumber(null))
    }

    @Test
    fun wholePositiveNumberShouldPassIfEmpty() {
        assertTrue(ModelUtils.isWholePositiveNumber(""))
    }

    @Test
    fun wholePositiveNumberShouldPass() {
        assertTrue(ModelUtils.isWholePositiveNumber("123"))
    }

    @Test
    fun wholePositiveNumberShouldPassWithLeadingZero() {
        assertTrue(ModelUtils.isWholePositiveNumber("000"))
    }

    @Test
    fun wholePositiveNumberShouldFailIfNegative() {
        assertFalse(ModelUtils.isWholePositiveNumber("-1"))
    }

    @Test
    fun wholePositiveNumberShouldFailIfLetters() {
        assertFalse(ModelUtils.isWholePositiveNumber("1a"))
    }

    @Test
    fun normalizeSameCenturyShouldPass() {
        val now = Calendar.getInstance()
        val year = 1997
        now.set(Calendar.YEAR, year)
        assertEquals(year, ModelUtils.normalizeYear(97, now))
    }

    @Test
    fun normalizeDifferentCenturyShouldFail() {
        val now = Calendar.getInstance()
        val year = 1997
        now.set(Calendar.YEAR, year)
        assertNotEquals(2097, ModelUtils.normalizeYear(97, now))
    }
}
